link.py

```diff
@@ -0,0 +1,14 @@
+from module import XMPPModule
+import halutils
+import re, requests
+ 
+class Link(XMPPModule):
+    
+    def handleMessage(self, msg):
+        obj = re.match('.*(http[s]?://.*)+', msg['body'])
```

Comment:
This might cause the regex to be recompiled on each message received, suggest storing as a class attribute, so is compiled once during `init()`.